core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -248,6 +256,20 @@ public void closeScanners() throws CrudException {
     }
   }
 
+  private void putIntoReadSetInSnapshot(Snapshot.Key key, Optional<TransactionResult> result) {
+    // In read-only mode, we don't need to put the result into the read set
+    if (!readOnly) {
+      snapshot.putIntoReadSet(key, result);
+    }
+  }
+
+  private void verifyNoOverlap(Scan scan, Map<Snapshot.Key, TransactionResult> results) {
```

Comment:
Also, we don't need to verify the overlap in read-only mode, since there are no write operations.